ci-operator/step-registry/origin/e2e/test/origin-e2e-test-commands.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+cluster_profile=/var/run/secrets/ci.openshift.io/cluster-profile
+export AWS_SHARED_CREDENTIALS_FILE=${cluster_profile}/.awscred
+export AZURE_AUTH_LOCATION=${cluster_profile}/osServicePrincipal.json
+export GCP_SHARED_CREDENTIALS_FILE=${cluster_profile}/gce.json
+export HOME=/tmp/home
+export PATH=/usr/libexec/origin:$PATH
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+mkdir -p "${HOME}"
+
+# if the cluster profile included an insights secret, install it to the cluster to
+# report support data from the support-operator
+if [[ -f "${cluster_profile}/insights-live.yaml" ]]; then
+    oc create -f "${cluster_profile}/insights-live.yaml" || true
+fi
+
+# set up cloud-provider-specific env vars
+KUBE_SSH_BASTION="$( oc --insecure-skip-tls-verify get node -l node-role.kubernetes.io/master -o 'jsonpath={.items[0].status.addresses[?(@.type=="ExternalIP")].address}' ):22"
+KUBE_SSH_KEY_PATH=${cluster_profile}/ssh-privatekey
+export KUBE_SSH_BASTION KUBE_SSH_KEY_PATH
+case "${CLUSTER_TYPE}" in
+gcp)
+    export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
+    export KUBE_SSH_USER=core
+    mkdir -p ~/.ssh
+    cp "${cluster_profile}/ssh-privatekey" ~/.ssh/google_compute_engine || true
+    export TEST_PROVIDER='{"type":"gce","region":"us-east1","multizone": true,"multimaster":true,"projectid":"openshift-gce-devel-ci"}'
+    ;;
+aws)
+    mkdir -p ~/.ssh
+    cp "${cluster_profile}/ssh-privatekey" ~/.ssh/kube_aws_rsa || true
+    export PROVIDER_ARGS="-provider=aws -gce-zone=us-east-1"
+    # TODO: make openshift-tests auto-discover this from cluster config
+    REGION="$(oc get -o jsonpath='{.status.platformStatus.aws.region}' infrastructure cluster)"
+    ZONE="$(oc get -o jsonpath='{.items[0].metadata.labels.failure-domain\.beta\.kubernetes\.io/zone}' nodes)"
+    export TEST_PROVIDER="{\"type\":\"aws\",\"region\":\"${REGION}\",\"zone\":\"${ZONE}\",\"multizone\":true,\"multimaster\":true}"
+    export KUBE_SSH_USER=core
+    ;;
+azure4) export TEST_PROVIDER=azure;;
+*) echo >&2 "Unsupported cluster type '${CLUSTER_TYPE}'"; exit 1;;
+esac
+
+mkdir -p /tmp/output
+cd /tmp/output
+
+if [[ "${CLUSTER_TYPE}" == gcp ]]; then
+    pushd /tmp
+    curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-256.0.0-linux-x86_64.tar.gz
+    tar -xzf google-cloud-sdk-256.0.0-linux-x86_64.tar.gz
+    export PATH=$PATH:/tmp/google-cloud-sdk/bin
+    mkdir gcloudconfig
+    export CLOUDSDK_CONFIG=/tmp/gcloudconfig
+    gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+    gcloud config set project openshift-gce-devel-ci
+    popd
+fi
+
+test_suite=openshift/conformance/parallel
+if [[ -e "${SHARED_DIR}/test-suite.txt" ]]; then
+    test_suite=$(<"${SHARED_DIR}/test-suite.txt")
+fi
+
+openshift-tests run "${test_suite}" \
+    --provider "${TEST_PROVIDER}" \
```

Comment:
This must be conditional based on whether TEST_PROVIDER is set.  If TEST_PROVIDER was not set, this flag should not be set.